build.gradle

```diff
@@ -2057,7 +2072,7 @@ allprojects {
 
     // All of our projects are java projects
 
-    apply plugin: "java"
+    apply plugin: 'java-library'
```

Comment:
What is the purpose of this change? This affects more than just the maven publishing, so I would like to make sure there are no implications to the rest of our build.

Reply:
As I explained in https://github.com/openjdk/jfx/pull/1232#issuecomment-1710607166 this changes the setup so that Gradle automatically generates the metadata (`pom` and `module` files) based on the "Model of the Software". Before the change, the metadata was completely constructed manually.

This change is needed in this PR to then extend the Model with the "Platform Variants" so that these are then also added to the metadata. (But it is in general a good improvement to let Gradle generate the metadata instead of to maintaining custom code for that).

In short you can say, all these changes pull things up one abstraction level. Instead of directly creating all metadata in the `publications {}` block, we give the information to Gradle in the "right place" so that it then has a complete and correct model of the JavaFX libraries and there relationships. From which it then creates the metadata when publishing.

This particular change here is needed, because since Gradle 4+, ["java-library" is used to represent Java libraries](https://docs.gradle.org/current/userguide/java_library_plugin.html). Which all of the JavaFX components are in Gradle terminology. This make the "api" scope for dependencies available we need further down.